src/EditorFeatures/CSharpTest/AutomaticCompletion/AutomaticBraceCompletionTests.cs

```diff
@@ -363,6 +363,100 @@ void M()
             CheckReturn(session.Session, 12, expectedAfterReturn);
         }
 
+        [WpfFact, Trait(Traits.Feature, Traits.Features.AutomaticCompletion)]
+        public void RecursivePattern_Nested()
+        {
+            var code = @"
+class C
+{
+    void M()
+    {
+        _ = this is { Name: $$ }
+    }
+}";
+
+            var expectedBeforeReturn = @"
+class C
+{
+    void M()
+    {
+        _ = this is { Name: { } }
+    }
+}";
+
+            var expectedAfterReturn = @"
+class C
+{
+    void M()
+    {
+        _ = this is { Name:
+            {
+
+            } }
+    }
+}";
+            using var session = CreateSession(code);
+            Assert.NotNull(session);
+
+            CheckStart(session.Session);
+            CheckText(session.Session, expectedBeforeReturn);
+            CheckReturn(session.Session, 16, expectedAfterReturn);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.AutomaticCompletion)]
+        public void RecursivePattern_Parentheses1()
+        {
+            var code = @"
+class C
+{
+    void M()
+    {
+        _ = this is { Name: $$ }
+    }
+}";
+            var expected = @"
+class C
+{
+    void M()
+    {
+        _ = this is { Name: () }
+    }
+}";
+
+            using var session = CreateSession(TestWorkspace.CreateCSharp(code), '(', ')');
+            Assert.NotNull(session);
+
+            CheckStart(session.Session);
+            CheckText(session.Session, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.AutomaticCompletion)]
+        public void RecursivePattern_Parentheses2()
+        {
+            var code = @"
+class C
+{
+    void M()
+    {
+        _ = this is { Name: { Length: (> 3) and $$ } }
+    }
+}";
+            var expected = @"
+class C
+{
+    void M()
+    {
+        _ = this is { Name: { Length: (> 3) and () } }
```

Comment:
does `(<` throw anything off due to `<` being a mismatched brace?

Reply:
It shouldn't because it's a normal binary operator. I need to figure out how to test it though.

Reply:
I wasn't able to make quick sense of this, so I filed #43560 as a follow-up.